test/e2e/namespace_annotation_test.go

```diff
@@ -0,0 +1,122 @@
+package e2e
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("[Basic] Backup/restore of 2 namespaces -  Namespace Annotation Test", func() {
+
+	client, err := newTestClient()
+	Expect(err).To(Succeed(), "Failed to instantiate cluster client for namespace annotation test")
+
+	BeforeEach(func() {
+		var err error
+		flag.Parse()
+		uuidgen, err = uuid.NewRandom()
+		Expect(err).To(Succeed())
+		if installVelero {
+			Expect(veleroInstall(context.Background(), veleroImage, veleroNamespace, cloudProvider, objectStoreProvider, false,
+				cloudCredentialsFile, bslBucket, bslPrefix, bslConfig, vslConfig, "")).To(Succeed())
+
+		}
+	})
+
+	AfterEach(func() {
+		if installVelero {
+			timeoutCTX, _ := context.WithTimeout(context.Background(), time.Minute)
+			err := veleroUninstall(timeoutCTX, client.kubebuilder, installVelero, veleroNamespace)
+			Expect(err).To(Succeed())
+		}
+
+	})
+
+	Context("When I create 2 namespaces", func() {
+		It("should be successfully backed up and restored including annotations", func() {
+			backupName := "backup-" + uuidgen.String()
+			restoreName := "restore-" + uuidgen.String()
+			fiveMinTimeout, _ := context.WithTimeout(context.Background(), 5*time.Minute)
+			RunNamespaceAnnotationTest(fiveMinTimeout, client, "nstest-"+uuidgen.String(), 2,
+				backupName, restoreName)
+		})
+	})
+})
+
+func RunNamespaceAnnotationTest(ctx context.Context, client testClient, nsBaseName string, numberOfNamespaces int, backupName string, restoreName string) error {
+	//timeout logic
+	shortTimeout, _ := context.WithTimeout(ctx, 5*time.Minute)
```

Comment:
You already got the 5-minute timeout context before passing it into this method, so I don't think it's needed here.